daemon/nvidia_linux.go

```diff
@@ -30,19 +33,28 @@ var allNvidiaCaps = map[nvidia.Capability]struct{}{
 }
 
 func init() {
-	if _, err := exec.LookPath(nvidiaHook); err != nil {
-		// do not register Nvidia driver if helper binary is not present.
-		return
-	}
-	capset := capabilities.Set{"gpu": struct{}{}, "nvidia": struct{}{}}
-	nvidiaDriver := &deviceDriver{
-		capset:     capset,
-		updateSpec: setNvidiaGPUs,
-	}
-	for c := range allNvidiaCaps {
-		nvidiaDriver.capset[string(c)] = struct{}{}
+	// Register Nvidia driver if Nvidia helper binary is present.
+	// Else, register AMD driver if AMD helper binary is present.
+	if _, err := exec.LookPath(nvidiaHook); err == nil {
+		capset := capabilities.Set{"gpu": struct{}{}, "nvidia": struct{}{}}
+		nvidiaDriver := &deviceDriver{
+			capset:     capset,
+			updateSpec: setNvidiaGPUs,
+		}
+		for c := range allNvidiaCaps {
+			nvidiaDriver.capset[string(c)] = struct{}{}
+		}
+		registerDeviceDriver("nvidia", nvidiaDriver)
+	} else if _, err := exec.LookPath(amdContainerRuntimeExecutableName); err == nil {
+		capset := capabilities.Set{"gpu": struct{}{}, "amd": struct{}{}}
+		amdDriver := &deviceDriver{
+			capset:     capset,
+			updateSpec: setAMDGPUs,
+		}
+		registerDeviceDriver("amd", amdDriver)
+	} else {
+		// no "gpu" capability
 	}
```

Comment:
Hm.. looks like the linter doesn't like this (even with a comment inside the branch 🤔)

```
daemon/nvidia_linux.go:55:9: SA9003: empty branch (staticcheck)
	} else {
	       ^
```